### DISCUSSION TITLE
Project Looks Interesting

### DISCUSSION THREAD

**Discussion by jeffolmstead** [lukethacoder/lwc-garden#8] [General]
I am going to pull this down and take a further look at it. I was looking for an alternative to LWR as the backend server for my lightning components and came upon this. Out of curiosity, have you ever hosted any sites off Salesforce? I am messing around with LWC OSS and they recently switched over to LWR as their default server. However, there are still a lot of pain points there as I have tried to set up a server with authentication (haven't yet succeeded on LWR, even when exposing the Express server). All this to say, I am curious, since you have this great tooling stack to make LWC easier and usable off salesforce, as to whether you have built productions sites around LWC and if so, was there a standard framework you started from? Thanks for any consideration you can share in this regard.

**Comment by lukethacoder** [comment]
Hey @jeffolmstead 

I have not deployed anything to production using the open source LWR setup. I don't see why it wouldn't work as their [documentation](https://developer.salesforce.com/docs/platform/lwr/guide/lwr-intro.html) outlines you can deploy to Heroku (or presumably any other node type host).

I wouldn't personally use LWR, not because it's not good, but because I find the greater JavaScript ecosystem to be much more powerful and open, and have much better communities building plugins and libraries. Something like [Next.js](https://nextjs.org/) or [Astro](https://astro.build/) would be more my flavour if I had to choose a framework off platform.

I could see benefits to reusing your on-platform LWCs off platform, which I assume is why SF provides an off-platform framework.

> Do note that some of the `@lwrjs/*` packages are not open sourced, and you may find it hard to get bugs fixed. Not to be confused with the [`lwc`](https://github.com/salesforce/lwc) framework which is fully open-sourced. There are a few LWR specific bugs that get raised here because there is no where else to raise them... but again, they usually get ignored or "handed to the LWR team" internally.
> 
> In contrast, a framework like Astro, given its all open sourced, you could either fix the bug yourself and submit a PR or raise a bug and it should be fixed much sooner.

TL:DR - I'd personally prefer not use LWR off platform, but there may be benefits of reusing on-platform built LWCs if built correctly.

**Reply by jeffolmstead** [reply]
@lukethacoder that is really good feedback, and yes, the major consideration is the benefit of reusing LWCs off platform and honestly to be able to use concepts developed off Saleforce platform back inside Digital Experience sites. While I am going to keep exploring LWR options, the Astro platform looks really interesting to me, and some of the concepts are similar to where it appears LWR is at with component hydration. I might try and see if LWC components themselves can drop into Astro due to it's lack of concern in what front end UI system is used (or so they claim). At a minimum I will learn something about Astro, at a maximum I will learn how tightly the LWC components are to LWR. Will keep following your work in this space and contribute back where I can.
